### PR TITLE
exit code improvements

### PR DESCRIPTION
**Pull request by vrothberg**
Please refer to the individual commits for change descriptions.  I am fairly confident it'll fix https://github.com/containers/podman/issues/14859 but lets see what CI (and long-term flakes) tell.

@mheon PTAL

No release notes needed as it's fixing unreleased issue.
```release-note
None
```

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
All kinds of test unhappiness @vrothberg

File: libpod/container_api.go [hunk 1]
```diff
@@ -551,6 +551,10 @@ func (c *Container) WaitForExit(ctx context.Context, pollInterval time.Duration)
 
 		exitCode, err := c.runtime.state.GetContainerExitCode(id)
 		if err != nil {
+			if errors.Is(err, define.ErrNoSuchExitCode) && c.ensureState(define.ContainerStateConfigured) {
```
[the remark this answers is not in the document]
Comment (vrothberg) [reply libpod/container_api.go:554]: Good idea!

**Comment by rhatdan** [comment]
LGTM

**Pull request merged**